src/main/java/am/ik/yavi/constraint/time/ComparableTemporalConstraintBase.java

```diff
@@ -0,0 +1,100 @@
+package am.ik.yavi.constraint.time;
+
+import am.ik.yavi.constraint.base.ConstraintBase;
+import am.ik.yavi.core.Constraint;
+import am.ik.yavi.core.ConstraintPredicate;
+
+import java.time.temporal.Temporal;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+import static am.ik.yavi.core.NullAs.VALID;
+import static am.ik.yavi.core.ViolationMessage.Default.*;
+
+/**
+ * This is the base class for constraints on Temporal classes that can be compared with
+ * others. Methods in the class require the {@link V} to extend Temporal and implement
+ * Comparable.
+ *
+ * @author Diego Krupitza
+ */
+abstract class ComparableTemporalConstraintBase<T, V extends Temporal & Comparable, C extends Constraint<T, V, C>>
+		extends ConstraintBase<T, V, C> {
+
+	/**
+	 * Is the given temporal before {@code other}
+	 *
+	 * @param other the other temporal that is after
+	 */
+	public C before(V other) {
+		this.predicates().add(ConstraintPredicate.of(x -> x.compareTo(other) <= -1,
```

Comment:
To reduce duplicated code

```java
return this.before(() -> other);
```

would be better?

Reply:
Yeah looks good to me. 
Remove duplicated code